packages/wonder-blocks-core/components/provider-group.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import * as React from "react";
+
+type Provider<P> = {|
+    type: React.ComponentType<P>,
+    props?: P,
+|};
+
+type Props = {|
+    providers: Array<Provider<any>>,
+    children: (Array<any>) => React.Node,
+|};
```

Comment:
I think it's possible to make the types more precise using https://flow.org/en/docs/types/utilities/#toc-tuplemap.